### ISSUE TITLE
!!/approve on github sometimes doesn't

### ISSUE THREAD

**Issue by tripleee** [Charcoal-SE/SmokeDetector#2965]
### What problem has occurred? What issues has it caused?

I added an `!!/approve` comment on several open PRs here on GitHub. Still PullApprove says "Pending: approval required by one of ...."

I deleted the failed comments and proceeded to approve from chat instead:
#2963 #2957 #2955 #2952 

Weirdly, #2951 looks like it was successfully approved, but still not merged. I will proceed to merge that manually.

### What would you like to happen/not happen?

Approving on Github should work just like approving from chat.

**Comment by makyen** [comment]
I agree that it would be nice if a `!!/approve` comment on GitHub did the same job of approving and merging, with resolving merge conflicts, that a `!!/approve PR#` does in chat. However, I've never seen it work fully the same (i.e. I've never seen it resolve the merge conflicts we typically get when another `!!/watch` or `!!/approve` has been done after the PR was created). It used to automatically merge when there were no conflicts. However, I haven't seen it do that in quite some time.

**Comment by angussidney** [comment]
If I remember correctly, before `!!/approve` on GitHub broke, it was using a merge strategy which automatically resolves most of the normal merge conflicts that we get (as a result single-line additions). The word onion rings a bell, but I feel like that's not right (a quick search of the git docs doesn't show up any results. Octopus maybe?)

**Comment by double-beep** [comment]
When `!!/approve`ing in GitHub by group 'code-admins' as set in `.pullapprove.yml`, then, a check named 'code-review/pullapprove' will be marked as 'passed'. However, the PR is not merged automatically (cc @makyen @angussidney), so it isn't broken at all.

It was suggested to trigger an automatic merge if it had a certain number of approving reviews, but unfortunately rejected. It was issue pullapprove/support#9. It was mentioned that a bot named automerge could do that ([GH repo](https://github.com/dropseed/automerge)) after adding an 'automerge' tag. However, I don't really like this idea.

I also found https://github.com/dgmltn/EggTimer from [this](https://stackoverflow.com/a/40398678) Stack Overflow answer which *may* help, I haven't tested it, though.

**Comment by ArtOfCode-** [comment]
@double-beep metasmoke _should_ automatically merge PRs once they're approved by PullApprove. Sounds like there's a bug there somewhere.

**Comment by iBug** [comment]
I have a feeling that a past PR (from me) that removed the creation of `.git/attributes` on MS side might be the cause, but my Git-fu told me [this `.gitattributes` file](https://github.com/Charcoal-SE/SmokeDetector/blob/master/.gitattributes) should always work.
